core/templates/pages/topic-editor-page/services/topic-editor-routing.service.spec.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for TopicEditorRoutingService.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+require('pages/topic-editor-page/services/topic-editor-routing.service.ts');
+
+describe('Topic editor routing service', function() {
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  var $rootScope = null;
+  var $location = null;
+  var $window = null;
+  var TopicEditorRoutingService = null;
+
+  beforeEach(angular.mock.inject(function($injector) {
+    $rootScope = $injector.get('$rootScope');
+    $location = $injector.get('$location');
+    $window = $injector.get('$window');
+    TopicEditorRoutingService = $injector.get('TopicEditorRoutingService');
+  }));
+
+  it('should return the default active tab name', function() {
+    expect(TopicEditorRoutingService.getActiveTabName()).toEqual('main');
+  });
+
+  it('should navigate to different tabs', function() {
+    expect(TopicEditorRoutingService.getActiveTabName()).toEqual('main');
+
+    TopicEditorRoutingService.navigateToSubtopicPreviewTab(1);
+    $rootScope.$apply();
+    expect(
+      TopicEditorRoutingService.getActiveTabName()).toEqual('subtopic_preview');
+
+    TopicEditorRoutingService.navigateToSubtopicEditorWithId(1);
+    $rootScope.$apply();
+    expect(
+      TopicEditorRoutingService.getActiveTabName()).toEqual('subtopic_editor');
+
+    TopicEditorRoutingService.navigateToQuestionsTab();
+    $rootScope.$apply();
+    expect(TopicEditorRoutingService.getActiveTabName()).toEqual('questions');
+
+    TopicEditorRoutingService.navigateToMainTab();
+    $rootScope.$apply();
+    expect(TopicEditorRoutingService.getActiveTabName()).toEqual('main');
+  });
+
+  it('should handler calls with unexpect paths', function() {
```

Comment:
*handle

Also, is this test correct? `$location.path()` just returns the path and does not set anything right?

Reply:
No it set's the url also. https://stackoverflow.com/questions/22515584/location-path-vs-location-hash-in-angularjs

Reply:
Hm, that contradicts what is written [here](https://docs.angularjs.org/api/ng/service/$location#path).
see
> Return path of current URL when called without any parameter.
> Change path when called with parameter and return $location.